src/Misc.pm

```diff
@@ -3988,6 +3990,16 @@ sub avoidList_near {
 	for my $player (@$playersList) {
 		my $avoidPlayer = $avoid{Players}{lc($player->{name})};
 		my $avoidID = $avoid{ID}{$player->{nameID}};
+
+		# Exit if the player is not on the avoid list
+		last if (!$avoidPlayer and !$avoidID);
```

Comment:
This is wrong. The first player not on `playersList` breaks the avoid function for all other users. This is supposed to be `next`, not `last`

Reply:
yes, you are absolutely right!